### PR TITLE
Build without path and sort

### PR DESCRIPTION
**Pull request by radarsat1**
These two patches make the build work if I remove folders externals/PATH_SDK and externals/sort.  These are the two main license-related pain points for #236.  Please review.

### CODE REVIEW

File: numerics/src/tools/NumericsSparseMatrix.c [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #include "NumericsSparseMatrix.h"
 #include "numerics_verbose.h"
 #include "NumericsMatrix.h"
+#include "string.h" // memcpy
```
Comment (xhub) [inline numerics/src/tools/NumericsSparseMatrix.c]: shouldn't that be #include <string.h>

File: numerics/src/tools/NumericsSparseMatrix.c [hunk 2]
```diff
@@ -44,10 +45,20 @@ typedef struct
...
 } sort_indices_struct;
 
+#ifdef HAVE_SORT
 #define SORT_NAME sorter
 #define SORT_TYPE sort_indices_struct
 #define SORT_CMP(x, y) ((x).i - (y).i)
 #include "sort.h"
+#else
+#include "stdlib.h" // qsort
+static int sort_indices_struct_cmp(const void *a, const void *b)
```
Comment (xhub) [inline numerics/src/tools/NumericsSparseMatrix.c]: ditto

File: numerics/src/tools/NumericsSparseMatrix.c [hunk 3]
```diff
@@ -44,10 +45,20 @@ typedef struct
...
+{
+  const sort_indices_struct *sa = (const sort_indices_struct *) a;
+  const sort_indices_struct *sb = (const sort_indices_struct *) b;
+  return (sa->i > sb->i) - (sa->i < sb->i);
+}
```
Comment (xhub) [inline numerics/src/tools/NumericsSparseMatrix.c]: I was just curious about that magic: is sa->i - sb->i not sufficient? If my C99 is not too rusty, sa->i > sb->i evaluates to a boolean. So the result is a difference of boolean promoted to int. It is a bit confusing for me. I have to admit that I don't remember that piece of code very well ...

**Comment by radarsat1** [comment]
Thanks, quite right about the brackets, will fix.  The sort comparison code is adapted straight from the documentation, the idea is that it's comparing two non-inclusive inequalities so if they're less it's -1, greater it's 1, and equal it's 0.  (like strcmp)  It should be correct I think.

**Comment by radarsat1** [comment]
This documentation that is: https://www.gnu.org/software/libc/manual/html_node/Comparison-Functions.html#Comparison-Functions

**Pull request merged**